documentation/CAMARA-OIDC-profile.md

```diff
@@ -0,0 +1,1020 @@
+
+    * **_invalid_client_**
+
+      Client authentication failed (e.g., unknown client, no client authentication included, or unsupported authentication method). The authorization server MAY return an HTTP 401 (Unauthorized) status code to indicate which HTTP authentication schemes are supported. If the client attempted to authenticate via the `Authorization` request header field, the authorization server MUST respond with an HTTP 401 (Unauthorized) status code and include the `WWW-Authenticate` response header field matching the authentication scheme used by the client.
+
+    * **_invalid_grant_**
+
+      The provided authorization grant authorization code is invalid, expired, revoked, does not match the redirection URI used in the authorization request, or was issued to another client.
+
+    * **_unauthorized_client_**
+
+      The authenticated client is not authorized to use this authorization grant type.
+
+    * **_unsupported_grant_type_**
+
+      The authorization grant type is not supported by the authorization server.
+
+    * **_invalid_scope_**
+
+      The requested scope is invalid, unknown, malformed, or exceeds the scope granted by the resource owner.
+
+
+* **error_description**
+
+  OPTIONAL. Human-readable ASCII text providing additional information, used to assist the client developer in understanding the error that occurred. Values for the `error_description` parameter MUST NOT include characters outside the set %x20-21 / %x23-5B / %x5D-7E.
+
+
+The parameters are included in the entity-body of the HTTP response using the `application/json` media type.
+
+`error_description` parameter is not a valid field to take decisions on the developer side, as its value may vary depending on the server implementation, language, format or context. Therefore, the developer should base their logic only on the `error` parameter.
+
+The following is a non-normative example Token Error Response:
+
+```http
+HTTP/1.1 400 Bad Request
+Content-Type: application/json
+```
+```json
+{
+  "error": "unauthorized_client"
+}
+```
+
+## The Scope Parameter
+
+Oauth scope values determine the specific CAMARA services being requested by the Service Provider, subject to the SP being registered to use those services.
+
+The Open Gateway Authorization Request MUST contain the `scope` parameter, which is a space delimited, case-sensitive list of ASCII strings (scope values). Multiple scope values can be requested simultaneously, subject to the SP being registered to use those “scopes”.
+
+For OpenID requests, the scope values MUST include `openid`, to indicate that the request is an OpenID Connect request, followed by other values depending on the specific CAMARA services being requested.
+
+A purpose must be declared within the requested scope for the following grant types:
```

Comment:
You are right, but a purpose is not the same as a scope. The purpose is the reason why you are accessing the data (e.g. "I want a token to access the 'sim-swap' scope for the purpose 'dpv:FraudManagement'").

As I say in https://github.com/camaraproject/IdentityAndConsentManagement/pull/113/files#r1470035920 the current approach to convey the purpose and its format was a short-term agreement. There were other options on the table (e.g. sending the purpose in a separate "purpose" parameter). We need to work on a better solution.

Reply:
I fully agree, the purpose is not the same as a scope. It is the reason for calling an API.
I concur palmerabollo's analysis: let's go for a separate purpose parameter. If then an authorization server does not understand it, it can ignore it.